src/pages/namespace/Settings/Variables/Create.tsx

```diff
@@ -5,47 +5,58 @@ import {
   DialogHeader,
   DialogTitle,
 } from "~/design/Dialog";
+import Editor, { EditorLanguagesType } from "~/design/Editor";
+import MimeTypeSelect, {
+  MimeTypeType,
+  mimeTypeToLanguageDict,
+} from "./MimeTypeSelect";
 import { SubmitHandler, useForm } from "react-hook-form";
 import { VarFormSchema, VarFormSchemaType } from "~/api/variables/schema";
+import { useEffect, useState } from "react";
 
 import Button from "~/design/Button";
 import { Card } from "~/design/Card";
-import Editor from "~/design/Editor";
 import FormErrors from "~/componentsNext/FormErrors";
 import Input from "~/design/Input";
-import MimeTypeSelect from "./MimeTypeSelect";
 import { PlusCircle } from "lucide-react";
-import { useState } from "react";
 import { useTheme } from "~/util/store/theme";
 import { useTranslation } from "react-i18next";
 import { useUpdateVar } from "~/api/variables/mutate/updateVariable";
 import { zodResolver } from "@hookform/resolvers/zod";
 
 type CreateProps = { onSuccess: () => void };
 
+const defaultMimeType: MimeTypeType = "application/json";
+
 const Create = ({ onSuccess }: CreateProps) => {
   const { t } = useTranslation();
   const theme = useTheme();
 
   const [name, setName] = useState<string | undefined>();
   const [body, setBody] = useState<string | undefined>();
-  const [mimeType, setMimeType] = useState<string | undefined>();
+  const [mimeType, setMimeType] = useState<MimeTypeType>(defaultMimeType);
+  const [editorLanguage, setEditorLanguage] =
+    useState<EditorLanguagesType>("plaintext");
 
   const {
     handleSubmit,
     formState: { errors },
   } = useForm<VarFormSchemaType>({
     resolver: zodResolver(VarFormSchema),
-    // mimeType defaults to text/plain to avoid backend defaulting to
+    // mimeType should always be initialized to avoid backend defaulting to
     // "text/plain, charset=utf-8", which does not fit the options in
     // MimeTypeSelect
     values: {
       name: name ?? "",
       content: body ?? "",
-      mimeType: mimeType ?? "text/plain",
+      mimeType: mimeType ?? "application/json",
     },
   });
 
+  useEffect(() => {
+    setEditorLanguage(mimeTypeToLanguageDict[mimeType]);
+  }, [mimeType]);
+
```

Comment:
I would suggest removing this `useEffect` here and introduce something like a `onMimetypeChange` function that takes care of changing both variables.

```typescript
const onMimetypeChange = (value: MimeTypeType) => {
    setMimeType(value);
    setEditorLanguage(mimeTypeToLanguageDict[value]);
};
``` 
and wire this function to the MimeTypeSelect component.
```typescript
<MimeTypeSelect
  id="mimetype"
  mimeType={mimeType}
  onChange={onMimetypeChange}
/>
``` 

### Why?
In my understanding, `useEffect` should only be used to react to a change that is not controlled by the component explicitly or happens asynchronously (like waiting for data to be fetched), but in this case the effect reacts to a `mimeType` change which is fully controlled by this component.

Reply:
I updated this as suggested and also made similar changes in the Edit component.

While doing this, I also updated the defaults for mimeType / language. 

They are now intentionally different in Create and Edit. In Create, for new variables, the default is json, like in the old UI. 

In Edit, I renamed the variable to fallbackMimeType and set it to text/plain. This takes effect when an existing variable has a mime type that is not implemented, like for example, XML. This can happen when you upload a file (in the old UI, then the extension determines the mime type).

In DIR-624, we might find a better solution for how to deal with file uploads that result in unsupported mimeTypes.